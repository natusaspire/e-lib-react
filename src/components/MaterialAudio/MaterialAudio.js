import React, { memo } from 'react';

const MaterialAudio = ({ material }) => (
  <div className="card">
    <div className="card__header">{material.title}</div>
    <div className="content">
      <div className="card__player">
        <audio controls src={material.url}>
          Sorry, your browser does not support embedded audio, you can
          {' '}
          <b>
            <a className="card__link" href={material.url}>download</a>
          </b>
          {' '}
          it.
        </audio>
      </div>
    </div>
  </div>
);

export default memo(MaterialAudio);
