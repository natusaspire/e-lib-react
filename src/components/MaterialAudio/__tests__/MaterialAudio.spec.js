import React from 'react';
import { shallow } from 'enzyme';

import MaterialAudio from '@/components/MaterialAudio';

describe('<MaterialAudio />', () => {
  it('should render audio material', () => {
    const material = {
      id: 1,
      type: 'audio',
      title: 'Material title',
      url: 'audio.mp3'
    };

    const wrapper = shallow(<MaterialAudio material={material} />);

    expect(wrapper).toMatchSnapshot();
  });
});
