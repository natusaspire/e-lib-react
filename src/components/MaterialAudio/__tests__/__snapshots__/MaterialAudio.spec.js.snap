// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MaterialAudio /> should render audio material 1`] = `
<div
  className="card"
>
  <div
    className="card__header"
  >
    Material title
  </div>
  <div
    className="content"
  >
    <div
      className="card__player"
    >
      <audio
        controls={true}
        src="audio.mp3"
      >
        Sorry, your browser does not support embedded audio, you can
         
        <b>
          <a
            className="card__link"
            href="audio.mp3"
          >
            download
          </a>
        </b>
         
        it.
      </audio>
    </div>
  </div>
</div>
`;
